src/mapgen/mapgen_fractal.cpp

```diff
@@ -391,24 +403,29 @@ s16 MapgenFractal::generateTerrain()
 	s16 stone_surface_max_y = -MAX_MAP_GENERATION_LIMIT;
 	u32 index2d = 0;
 
-	noise_seabed->perlinMap2D(node_min.X, node_min.Z);
+	if (spflags & MGFRACTAL_TERRAIN)
```

Comment:
Also here. I know that `spflags` are supposed to be unmodified during modification, but forgetting the right checks here could cause issues that are hard to find.

Reply:
Ok.

Reply:
Done.